src/app.ts

```diff
@@ -84,7 +84,7 @@ app.use(function (req, res, next) {
             req.secure || req.get('X-Forwarded-Proto') === 'https'
 
         if (!isRequestSsl) {
-            const newUrl = `https://${req.get('host')}${req.originalUrl}`
+            const newUrl = `https://${req.hostname}:${CaptainConstants.configs.nginxPortNumber80}${req.originalUrl}`
```

Comment:
This line turned out to introduced a bug. Fixed in https://github.com/caprover/caprover/commit/017bcca3c8def526349e7e092cdcf5ea5b4045fb


Basically, it would redirect to `https://captain.domain.com:80` - and port 80 is not listening for SSL connections.